src/components/App.tsx

```diff
@@ -9,18 +8,33 @@ import './styles/App.scss';
 
 import Header from './Header';
 import Home from './Home';
-import Level1 from './Level1';
+import { LEVELS } from './Levels';
 
 export const FirebaseClassContext = createContext(new _Firebase());
 
 export default function App(): JSX.Element {
-  const [isSignedIn, setIsSignedIn] = useState(false);
-  const [_firebase] = useState<_Firebase>(new _Firebase());
+  const [isSignedIn, setIsSignedIn] = useState<boolean | undefined>(undefined);
+  const _firebase = useContext(FirebaseClassContext);
 
   useEffect(() => {
-    _firebase.load({ setIsSignedIn });
+    _firebase.load(
+      () => {
+        setIsSignedIn(true);
+      },
+      () => {
+        setIsSignedIn(false);
+      },
+    );
   }, []);
 
+  if (isSignedIn === undefined) {
+    return (
+      <div>
+        <h1>Loading...</h1>
+      </div>
+    );
```

Comment:
note to self.. make this a component in the another PR

ill make an issue so we dont forget